src/Akka.Persistence.Sql.Hosting/HostingExtensions.cs

```diff
@@ -0,0 +1,252 @@
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when both <paramref name="journalOptionConfigurator"/> and <paramref name="snapshotOptionConfigurator"/> are null.
+        /// </exception>
+        public static AkkaConfigurationBuilder WithSqlPersistence(
+            this AkkaConfigurationBuilder builder,
+            Action<SqlJournalOptions>? journalOptionConfigurator = null,
+            Action<SqlSnapshotOptions>? snapshotOptionConfigurator = null,
+            bool isDefaultPlugin = true)
+        {
+            if (journalOptionConfigurator is null && snapshotOptionConfigurator is null)
+                throw new ArgumentException($"{nameof(journalOptionConfigurator)} and {nameof(snapshotOptionConfigurator)} could not both be null");
+            
+            SqlJournalOptions? journalOptions = null;
+            if (journalOptionConfigurator is { })
+            {
+                journalOptions = new SqlJournalOptions(isDefaultPlugin);
+                journalOptionConfigurator(journalOptions);
+            }
+
+            SqlSnapshotOptions? snapshotOptions = null;
+            if (snapshotOptionConfigurator is { })
+            {
+                snapshotOptions = new SqlSnapshotOptions(isDefaultPlugin);
+                snapshotOptionConfigurator(snapshotOptions);
+            }
+
+            return builder.WithSqlPersistence(journalOptions, snapshotOptions);
+        }
+        
+        /// <summary>
+        ///     Adds Akka.Persistence.SqlServer support to this <see cref="ActorSystem"/>. At least one of the options
+        ///     have to be populated else this method will throw an exception.
+        /// </summary>
+        /// <param name="builder">
+        ///     The builder instance being configured.
+        /// </param>
+        /// <param name="journalOptions">
+        ///     <para>
+        ///         An instance of <see cref="SqlJournalOptions"/>, used to configure the journal plugin
+        ///     </para>
+        ///     <i>Default</i>: <c>null</c>
+        /// </param>
+        /// <param name="snapshotOptions">
+        ///     <para>
+        ///         An instance of <see cref="SqlSnapshotOptions"/>, used to configure the snapshot store plugin
+        ///     </para>
+        ///     <i>Default</i>: <c>null</c>
+        /// </param>
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when both <paramref name="journalOptions"/> and <paramref name="snapshotOptions"/> are null.
+        /// </exception>
+        public static AkkaConfigurationBuilder WithSqlPersistence(
+            this AkkaConfigurationBuilder builder,
+            SqlJournalOptions? journalOptions = null,
```

Comment:
Won't this configuration be 100% ineffective if a connection string isn't specified? Shouldn't we disallow that on the API?

Reply:
This is the "advanced" version of the API, when people need to modify all aspect of their database mapping through the option classes, we already have a convenience easy to use method on line 75 of this file.

Reply:
Connection string checking is done inside the options to HOCON transform, the method will throw on missing connection string or provider name

Reply:
The API design/pattern is the same as what we have for SqlServer and PostgreSql so that all of our plugin have a similar API pattern